Client/Coordinators/Router/Route.swift

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// Enumeration representing different navigational routes in an application.
+enum Route {
+    /// Represents a search route that takes a URL string and a boolean value indicating whether the search is private or not.
+    case search(url: String, isPrivate: Bool)
```

Comment:
We need other type of searches too, so it could be a SearchAction sub-enum of some sort:
- For the search (which I think is our switchToTabForURLOrOpen) we sometimes pass in a uuid as well on top of the url and isPrivate parameters
- openBlankNewTab with parameters: focusLocationField, isPrivate
- openText which opens a new tab and submit text on the URL bar (and select it to enter overlay mode)

We might need others too, but starting with those covers most cases we'll need.

Reply:
Agree. I'm adding them to the next PR with the decoding part.